### PR TITLE
#39 rename module

### LINKED ISSUE #39: Rename root module
Currently all modules are located in the same namespace. To avoid polluting the global module namespace we need to move all modules to a sub-module, e.g. `luasql.exasol`.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #39

### CODE REVIEW

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -34,14 +34,14 @@ eval $(luarocks path)
 Then install `exasol-driver-lua` and required [Lua libraries](../../dependencies.md) by running
 
 ```sh
-luarocks install --local luasqlexasol
+luarocks install --local luasql-exasol
```
Comment (tomuben) [inline doc/user_guide/user_guide.md:37]: I assume it's correct, but on all other places it's 
`luasql.exasol`

So just wanted to mention it...

**Review by tomuben — APPROVED** [review]

**Review by tomuben — COMMENTED** [review]
CI test failed.

File: tools/format-lua.sh [hunk 2]
```diff
@@ -7,14 +7,15 @@ set -o pipefail
 base_dir="$( cd "$(dirname "$0")/.." >/dev/null 2>&1 ; pwd -P )"
 readonly base_dir
 
-readonly src_module_path="$base_dir/src"
+readonly src_module_path="$base_dir/src/"
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply tools/format-lua.sh]: 👍

File: doc/user_guide/user_guide.md (hunk 1 above)
Comment (kaklakariada) [reply doc/user_guide/user_guide.md:37]: Thanks! It's OK. We need `luasql.exasol` for loading modules and `luasql-exasol` will be the luarocks package name (it's not published yet). This follows the luasql naming conventions: https://keplerproject.github.io/luasql/

**Review by tomuben — APPROVED** [review]

**Pull request merged**